e2e/playwright.config.js

```diff
@@ -13,8 +13,8 @@ export default defineConfig({
   forbidOnly: !!process.env.CI,
   // Retry on CI only
   retries: process.env.CI ? 2 : 0,
-  // Opt out of parallel tests on CI.
-  workers: process.env.CI ? 1 : undefined,
+  // Parallelize tests.
+  workers: 10,
```

Comment:
Have you tried using nproc to set the workers to the number of processors rather than hardcoding? e.g. when running the playwright command doing something like:
```
npx playwright test --workers=$(nproc)
```

Reply:
@doshitan what do you think of this optimization? should we just hardcode it for now? seems reasonable, i think most of our laptops have 8 cores, and a quick google seems to show that github actions runners have 4 cores, so i think hardcoding to 8 seems reasonable. or if we wanted to make it more elegant we can add something to `e2e/run-e2e-test` to check for the number of cores like:

```
if [[ "$OSTYPE" == "linux-gnu"* ]]; then
  CORES=$(nproc)
elif [[ "$OSTYPE" == "darwin"* ]]; then
  CORES=$(sysctl -n hw.ncpu)
else
  CORES=1 # Fallback to running without parallelization
fi

npx playwright test --config "${APP_NAME}/playwright.config.js" --workers=$CORES "$@"
```